components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/common/Util.java

```
@@ -740,4 +748,109 @@ public static void safelyRemoveHandlers(ChannelPipeline pipeline, String... hand
+     * @return HttpCarbon message
+     */
+    public static HTTPCarbonMessage createInboundRespCarbonMsg(ChannelHandlerContext ctx,
+            HttpResponse httpResponseHeaders, HTTPCarbonMessage outboundRequestMsg) {
+        HTTPCarbonMessage inboundResponseMsg = new HttpCarbonResponse(httpResponseHeaders, new DefaultListener(ctx));
+        inboundResponseMsg.setProperty(Constants.POOLED_BYTE_BUFFER_FACTORY,
+                new PooledDataStreamerFactory(ctx.alloc()));
+
+        inboundResponseMsg.setProperty(Constants.DIRECTION, Constants.DIRECTION_RESPONSE);
+        inboundResponseMsg.setProperty(Constants.HTTP_STATUS_CODE, httpResponseHeaders.status().code());
+
+        //copy required properties for service chaining from incoming carbon message to the response carbon message
+        //copy shared worker pool
+        inboundResponseMsg.setProperty(Constants.EXECUTOR_WORKER_POOL, outboundRequestMsg
+                .getProperty(Constants.EXECUTOR_WORKER_POOL));
+
+        return inboundResponseMsg;
+    }
+
+    /**
+     * Check whether a connection should alive or not.
+     * @param keepAliveConfig of the connection
+     * @param outboundRequestMsg of this particular transaction
+     * @return true if the connection should be kept alive
+     */
+    public static boolean isKeepAlive(KeepAliveConfig keepAliveConfig, HTTPCarbonMessage outboundRequestMsg) {
+        switch (keepAliveConfig) {
+        case AUTO:
+            if (Float.valueOf((String) outboundRequestMsg.getProperty(Constants.HTTP_VERSION)) > Constants.HTTP_1_0) {
+                return true;
+            } else {
+                return false;
+            }
+        case ALWAYS:
+            return true;
+        case NEVER:
+            return false;
+        default:
+            // The execution will never reach here. To make the code compilable, default case has to be placed here.
+            return true;
```

If execution should never reach here, better throw an exception rather than returning true. Then if someone adds another value to the enum(mistakenly may be) it will be caught here.